lib/ReactViews/MapIconButton/MapIconButton.jsx

```diff
@@ -121,7 +121,7 @@ function MapIconButton(props) {
             medium
             css={`
               display: block;
-              transition: all 0.5s ease;
+              transition: all 0.3s ease;
```

Comment:
https://github.com/TerriaJS/terriajs/pull/4165#discussion_r401348661
I _think_ all you need here is width (even though transitioning on width is not ideal) - could you update this?

Reply:
Heads up: please excuse my lack of knowledge on css transitions 😬 

The reason why I didn't transition on width is because (from what I understand of transitions) I would need to set specific widths for all of the buttons to have them transition from 0px width to x pixels. I couldn't find another way to do this without setting widths for each button. Also I was following the way the invision buttons work (actual invision buttons, not the ones on the designs in invision), since that's what Vic wanted ideally.

Reply:
Might discuss a bit more on slack, and paste the outcome of discussion here once we've agreed on something

Reply:
Sorry to this note I meant change it from `all 0.3s ease` to something (example, include as many as you need) like `opacity 0.3s ease, max-width 0.3s ease` so that you aren't telling the browser to needlessly animate on every single thing possible. It's generally better to explicitly choose the properties you want animated instead of doing `all` as it's less performant for no upside. (And on top of this, transitioning on width isn't ideal because it causes browser to do layout changes instead of only repaints - but less important for a button like this, VS for example, a side panel that covers ~30% of the viewport)
More info:
https://developers.google.com/web/fundamentals/design-and-ux/animations/animations-and-performance
https://csstriggers.com/

Reply:
As discussed we can resolve this as part of https://github.com/TerriaJS/terriajs/issues/4281